src/app/components/datacard/datacard.component.ts

```diff
@@ -1,57 +1,116 @@
-import { Component, OnInit, Input, AfterViewInit } from "@angular/core"
+import { Component, OnInit, OnDestroy } from "@angular/core"
 import { CovidService } from "src/app/services/covid.service"
+import { ActivatedRoute } from "@angular/router"
+import { Subscription } from "rxjs"
+import { CountryApi, CountryReport, CountryReportAdapter } from "src/app/model/country-report.model"
 
 @Component({
     selector: "app-datacard",
     templateUrl: "./datacard.component.html",
     styleUrls: ["./datacard.component.scss"],
 })
-export class DatacardComponent implements OnInit {
-    private _showCountry: string = ""
+export class DatacardComponent implements OnInit, OnDestroy {
+    private suscription: Subscription
+    countries: CountryApi[]
+    countryReport: CountryReport
+
     confirmedCases: any
+    iniciar: boolean
+    showCountry: boolean
     loading: boolean
     error: boolean
-    warning: boolean = false
+    warning: boolean
     mensajeError: string
-    iniciar: boolean
+    summary: any[] = []
+
+    constructor(private covid: CovidService, private router: ActivatedRoute, private adapter: CountryReportAdapter) {}
+
+    ngOnInit(): void {
+        this.stateHandler("init")
+        this.suscription = this.covid.getCountriesSummary().subscribe((countries: CountryApi[]) => {
+            this.countries = countries
 
-    get showCountry(): string {
-        return this._showCountry
+            this.router.params.subscribe((params) => {
+                if (Object.keys(params).length !== 0) {
+                    this.confirmedCases = this.getCountrySummary(params.country)
+                    this.stateHandler("data")
+                } else if (Object.keys(params).length === 0) {
```

Comment:
You shouldn't need to use `else if` in this case, just else. If the first condition is false, in this case, you can assume that the second one will always be true.

Reply:
else if removed